### PR TITLE
Allow manually calling `search-api-v2`

### PR DESCRIPTION
**Pull request by csutter**
As a first step to building the integration with `search-api-v2`, allow `finder-frontend` to call it if and only if an explicit `use_v2` query parameter is supplied.

- Bump `gds-api-adapters` to get new adapter for `search-api-v2`
- Add `Services#search_api_v2`
- Update `Search::Query` to call `search-api-v2` if `use_v2` parameter is present and truthy (accoring to `ActiveModel::Type::Boolean`)

### CODE REVIEW

File: app/lib/search/query.rb [hunk 1]
```diff
@@ -99,5 +103,9 @@ def fetch_search_response(content_item)
         end
       end
     end
+
+    def use_v2_api?
+      ActiveModel::Type::Boolean.new.cast(filter_params["use_v2"])
```
Comment (leenagupte) [inline app/lib/search/query.rb:108]: This is nice!

File: Gemfile.lock [hunk 2]
```diff
@@ -145,7 +145,7 @@ GEM
     factory_bot (6.3.0)
       activesupport (>= 5.0.0)
     ffi (1.15.5)
-    gds-api-adapters (90.0.0)
+    gds-api-adapters (91.0.0)
```
Comment (leenagupte) [inline Gemfile.lock:148]: Not blocker: We'd normally put gem updates in a separate commit to increase readability and reduce noise in the commit.

**Review by leenagupte — APPROVED** [review]
👍

File: Gemfile.lock (hunk 2 above)
Comment (csutter) [reply Gemfile.lock:148]: Good shout – I can see Dependabot has already created #3153, I'll let you merge that and will rebase this PR!
